### PR TITLE
Update RHEL6 Remediations from templates

### PR DESCRIPTION
**Pull request by ybznek**
I have modified .csv files to generate the same set of templates.
I **EXPECT** that these functions work on rhel6. Does it need extra testing?

After that PR.
We should be able to delete these files and create them during build.

### CODE REVIEW

File: RHEL/6/input/oval/templates/kernel_modules_disabled.csv [hunk 1]
```diff
@@ -10,4 +10,4 @@ squashfs
 tipc
 udf
 usb-storage
-bluetooth
```
Comment (redhatrises) [inline RHEL/6/input/oval/templates/kernel_modules_disabled.csv]: @ybznek why was the bluetooth module removed?

File: RHEL/6/input/oval/templates/services_disabled.csv [hunk 2]
```diff
@@ -22,7 +22,6 @@ netconsole,,
 netfs,,
 nfs,nfs-utils,
 nfslock,nfs-utils,
-ntpdate,ntpdate,
```
Comment (redhatrises) [inline RHEL/6/input/oval/templates/services_disabled.csv]: @ybznek why was this removed?

File: RHEL/6/input/oval/templates/services_enabled.csv [hunk 3]
```diff
@@ -4,7 +4,6 @@ ip6tables,iptables-ipv6
 iptables,iptables
 irqbalance,irqbalance
 ntpd,ntp
-pcscd,pcsc-lite
```
Comment (redhatrises) [inline RHEL/6/input/oval/templates/services_enabled.csv]: @ybznek why was this removed?

File: RHEL/6/input/oval/templates/sysctl_ipv6_values.csv [hunk 4]
```diff
@@ -1,9 +1,4 @@
 # Add <sysctl_parameter_name, desired_value> to generate hard-coded OVAL and remediation content.
 # Add <sysctl_parameter_name,> to generate OVAL and remediation content that use the XCCDF value.
-net.ipv6.conf.all.accept_ra,
 net.ipv6.conf.default.accept_ra,
 net.ipv6.conf.default.accept_redirects,
-net.ipv6.conf.all.accept_source_route,
-net.ipv6.conf.default.accept_source_route,
-net.ipv6.conf.all.accept_redirects,
-net.ipv6.conf.all.forwarding,
```
Comment (redhatrises) [inline RHEL/6/input/oval/templates/sysctl_ipv6_values.csv]: @ybznek why were these removed?

File: RHEL/6/input/remediations/bash/templates/Makefile [hunk 5]
```diff
@@ -24,3 +24,7 @@ copy:
 clean:
 	rm -f output/*.sh
 	rm -rf output/
+
+clean_copied:
```
Comment (redhatrises) [inline RHEL/6/input/remediations/bash/templates/Makefile]: @ybznek I wonder if the build system should get the templates from `templates/output` instead of this target?

**Comment by ybznek** [comment]
@redhatrises 
I haven't investigated these rules, I understood that ssg build system read scripts from
`RHEL/6/input/remediations/bash`.

But when I run `make ; make copy` in `RHEL/6/input/remediations/bash/templates`
It will modify files in `RHEL/6/input/remediations/bash` and add some new scripts (kernel_module_bluetooth,...).

After my changes I expected, that we would be able to delete some of scripts `RHEL/6/input/remediations/bash` and generate them during build.

But without my changes it would create more files than was there before.

Am I wrong?

**Comment by ybznek** [comment]
If I haven't oversight it, remmediation for bluetooth module isn't in guide, either.

http://static.open-scap.org/ssg-guides/ssg-rhel6-guide-index.html

**Comment by iankko** [comment]
@ybznek 
> @redhatrises
I haven't investigated these lines in csv, I understood that ssg build system read scripts from
RHEL/6/input/remediations/bash.

The remediations are taken from two locations:
* ```shared/remediations/bash``` (for those that are shared across multiple products), and
* ```RHEL/6/input/remediations/bash``` (for those that on top of that are RHEL-6 specific. Couldn't be moved to shared since they are RHEL-6 specific, or weren't moved to shared folder yet, since the possibility if shared/ version could be used wasn't investigated yet).

> But when I run make ; make copy in RHEL/6/input/remediations/bash/templates
It will modify files in RHEL/6/input/remediations/bash and add some new scripts (kernel_module_bluetooth,...).

This IMHO is because:
* the content of CSV files is outdated (not synced),
* it doesn't count with the possibility of the remediations being stored also somewhere else (this Makefile has been created yet before concept of ```shared/``` folder was introduced).

> After my changes I expected, that we would be able to delete some of scripts RHEL/6/input/remediations/bash and generate them during build.

Correct. But to be able to do this it's necessary:
* First ensure content of CSV file for that remediation script to be up2date in ```shared/``` folder (make something truly generates only those *.sh files present in shared/remediations/bash folder),
* Ensure content of CSV file for that remediation type to be up2date in ```RHEL/6``` folder

Also, AFAIK ```make copy``` first runs ```make``` in the templates folder (for shared/ or RHEL/6 product, thus will make remediation scritps of all types, like package installed, package removed, service enabled / disabled, sysctl etc)

If you want to copy just one type of templates, you explicitly need to call that target to be generated (e.g. ```make sysctls``` to create remediations just for sysctl rules). Unfortunately I think right now it's not possible to use ```make copy``` to copy only generated remediations of one type.

Workaround is to use ```make sysctls``` and copy the output/*.sh files manually. Or modify the ```make copy``` to copy just one type.

> But without my changes it would create more files than was there before.

See above. More files are there, because ```make compare``` doesn't know about ```shared/remediations/bash folder and that some scripts might live there too).

> Am I wrong?

No. Just the issue of solving this isn't question of issue plain ```make copy``` and delete the existing remediations (like mentioned first ```make compare``` needs to be updated to get aware of shared/remediatons/bash locations. And compare the content also there). If it would be question of ```make copy``` we would generate the remediations on the fly already.

Sorry for being a courier of bad news.

**Comment by ybznek** [comment]
I will open new changed PR.
